pkg/controller/deployment/util/deployment_util.go

```diff
@@ -67,6 +67,10 @@ const (
 	RollbackTemplateUnchanged = "DeploymentRollbackTemplateUnchanged"
 	// RollbackDone is the done rollback event reason
 	RollbackDone = "DeploymentRollback"
+	// OverlapAnnotation marks deployments with overlapping selector with other deployments
+	// TODO: Delete this annotation when we no longer need to support a client
```

Comment:
In 1.7, we should probably check deployments with this annotation and clean it up, then remove it from our codebase in 1.8.